packages/rpc/src/HttpClient.ts

```diff
@@ -14,14 +33,65 @@ export default class HttpClient extends Client {
     });
 
     if (!res.ok) {
-      return { success: false, error: new Error(`HTTP error: ${res.status}`) };
+      return request.map((r) => ({
+        id: r.id,
+        success: false,
+        error: new Error(`HTTP error: ${res.status}`),
+      }));
     }
 
     try {
-      const result = (await res.json()) as unknown;
-      return { success: true, result };
+      const jsonRpcResponse = (await res.json()) as JsonRpcResponse[];
+      return jsonRpcResponse.map((r) => ({ id: r.id as string, success: true, result: r }));
     } catch (cause) {
-      return { success: false, error: new Error('Invalid JSON response', { cause }) };
+      return request.map((r) => ({
+        id: r.id,
+        success: false,
+        error: new Error('Invalid JSON response', { cause }),
+      }));
+    }
+  }
+
+  override sendRequest<const T extends RpcMethod>(
+    method: T,
+    ...params: RpcRequest[T]
+  ): Promise<RpcResult<T>> {
+    const deferred = deferredPromise<RpcResult<T>>();
+    const body = this.formatRequest(method, params);
+    this.requestMap.set(body.id, { deferred, body, method });
+    if (this.timer) clearTimeout(this.timer);
+    this.timer = setTimeout(() => this.sendBatch(), this.batchDuration);
+    return deferred.promise;
+  }
+
+  private async sendBatch() {
+    const clonedMap = new Map(this.requestMap);
+    this.requestMap.clear();
+    const requests = [...clonedMap.values()].map((item) => item.body);
+    const responses = await this.send(requests);
+
+    for (const request of requests) {
+      const response = responses.find((r) => r.id === request.id);
+      const clonedItem = clonedMap.get(request.id);
+      if (!clonedItem) {
```

Comment:
we should delete out any responses we handle. we can check afterwards if the map size is 0
```suggestion
      if (!clonedItem && !clonedMap.delete(response.id)) {
```